### PR TITLE
Initial change to add fields related to privatehub

### PR DESCRIPTION
**Pull request by johsun-msft**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/frontdoor.json [hunk 1]
```diff
@@ -1207,6 +1207,23 @@
           "description": "Location of the backend (IP address or FQDN)",
```
Comment (pichandwork) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/frontdoor.json]: >Location of the backend (IP address or FQDN) [](start = 26, length = 44)

Add "or private link resource Id"

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/examples/FrontdoorCreate.json [hunk 2]
```diff
@@ -66,20 +66,24 @@
             "properties": {
               "backends": [
                 {
+                  "type": "Public",
+                  "location": "westus",
```
Comment (pichandwork) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/examples/FrontdoorCreate.json:70]: >"location": "westus", [](start = 18, length = 21)

Isnt location valid only for private backend?

**Review by pichandwork — CHANGES_REQUESTED** [review]
:clock1:

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/examples/FrontdoorCreate.json (hunk 2 above)
Comment (johsun-msft) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/examples/FrontdoorCreate.json:70]: required for private, but allowed for public as well

**Review by ashnando1 — COMMENTED** [review]
Replace type with privateLinkId

Example is /subscriptions/0000000-0000-0000-0000-000000000000/resourceGroups/resourcegroup/providers/Microsoft.Network/privateLinkServices/PrivateLinkName

**Pull request merged**